.github/workflows/unittests.yml

```diff
@@ -33,18 +33,57 @@ jobs:
         ports:
         - 5672:5672
 
+    permissions:
+      packages: write
+
     steps:
     - uses: actions/checkout@v4
 
-    - name: Set up Python ${{matrix.python-version}}
-      uses: actions/setup-python@v5
+    - name: Set up Spack
+      uses: spack/setup-spack@v2
       with:
-        python-version: ${{matrix.python-version}}
+        buildcache: true  # Configure oci://ghcr.io/spack/github-actions-buildcache
+
+    - name: Install
+      env:
+        GITHUB_USER: ${{ github.actor }}
+        GITHUB_TOKEN: ${{ secrets.SPACK_OCI_TOKEN }}
+      run: |
+        sudo apt install gcc-11 g++-11 gfortran-11
+        which gcc-11
+        which g++-11
+        which gfortran-11
+        spack -e . add python@${{matrix.python-version}}
+        spack -e . install
+
+    - name: Run
+      shell: spack-bash {0}
+      run: |
+        spack env activate .
+        spack find
+
+    - name: Push packages and update index
+      env:
+        GITHUB_USER: agoscinski
+        GITHUB_TOKEN: ${{ secrets.SPACK_OCI_TOKEN }}
+      run: spack -e . buildcache push --base-image ubuntu:latest --update-index local-buildcache
+      if: ${{ !cancelled() }}
 
-    - name: install Python dependencies
+    - name: Install Python dependencies
+      shell: spack-bash {0}
       run: |
+        spack env activate .
+        which python
+        which pip
         pip install --upgrade pip 
         pip install hatch
 
     - name: run unittests
-      run: hatch test -py ${{matrix.python-version}} -n auto -v
+      shell: spack-bash {0}
+      run: |
+        spack env activate .
+        hatch test -py ${{matrix.python-version}} -n auto -v tests
+
+    - name: Setup tmate session
+      uses: mxschmitt/action-tmate@v3
+      if: ${{ failure() }}
```

Comment:
We can remove it, but I must say this is quite useful. The only annoying thing that one does not see immediately when a CI job failed because it is still runnnig keepen the the ssh connection open, and probably one should add the ssh key